airbyte-cdk/java/airbyte-cdk/typing-deduping/src/main/java/io/airbyte/integrations/base/destination/typing_deduping/DefaultTyperDeduper.java

```diff
@@ -239,6 +236,29 @@ public CompletableFuture<Optional<Exception>> typeAndDedupeTask(final StreamConf
     }, this.executorService);
   }
 
+  /**
+   * It can be expensive to build the errors array in the airbyte_meta column, so we first attempt an
+   * 'unsafe' transaction which assumes everything is typed correctly. If that fails, we will run a
+   * more expensive query which handles casting errors
+   *
+   * @param streamConfig the stream to type and dedupe
+   * @throws Exception if the safe query fails
+   */
+  private void typeAndDedupeTransactions(final StreamConfig streamConfig) throws Exception {
+    final String suffix = getFinalTableSuffix(streamConfig.id());
+    try {
+      LOGGER.info("Attempting typing and deduping for {}.{}", streamConfig.id().originalNamespace(), streamConfig.id().originalName());
+      final String unsafeSql = sqlGenerator.updateTable(streamConfig, suffix, minExtractedAtByStream.get(streamConfig.id()), false);
+      destinationHandler.execute(unsafeSql);
+      // TODO determine which Exceptions should not be retried even with safer sql
+    } catch (Exception e) {
+      LOGGER.error("Encountered Exception on unsafe SQL for stream %s %s, attempting with error handling"
```

Comment:
Nice logging

Reply:
🪵

Reply:
fyi the logger has built-in string formatting, e.g. `LOGGER.error("foo {} bar {}", "first param", "second param")`

Reply:
:doh: I for some reason thought that the override with variable inputs and auto-formatting would format the exception weirdly but it works fine